src/ray/raylet/node_manager.cc

```diff
@@ -1806,45 +1806,46 @@ void NodeManager::FinishAssignedTask(Worker &worker) {
   }
 }
 
-ActorTableData NodeManager::CreateActorTableDataFromCreationTask(
+std::shared_ptr<ActorTableData> NodeManager::CreateActorTableDataFromCreationTask(
     const TaskSpecification &task_spec) {
   RAY_CHECK(task_spec.IsActorCreationTask());
   auto actor_id = task_spec.ActorCreationId();
   auto actor_entry = actor_registry_.find(actor_id);
-  ActorTableData new_actor_data;
+  std::shared_ptr<ActorTableData> actor_info_ptr;
   // TODO(swang): If this is an actor that was reconstructed, and previous
   // actor notifications were delayed, then this node may not have an entry for
   // the actor in actor_regisry_. Then, the fields for the number of
   // reconstructions will be wrong.
   if (actor_entry == actor_registry_.end()) {
+    actor_info_ptr.reset(new ActorTableData());
```

Comment:
make_shared?

Reply:
One more copy.

Reply:
No matter to copy a smart pointer here I think.

Reply:
https://stackoverflow.com/questions/36477022/boost-initializing-shared-pointer-reset-vs-make-shared/36477600

In general, make_shared will be more efficient I think.